src/test/shared/filesystemUtilities.test.ts

```diff
@@ -30,8 +30,35 @@ describe('filesystemUtilities', () => {
             await remove(folder)
         }
     })
+    
+    describe('getNonexistentFilename()', () => {
+        it('failure modes', async () => {
+            assert.throws(() => {
+                getNonexistentFilename('/bogus/directory/', 'foo', '.txt', 99)
+            })
+            assert.throws(() => {
+                getNonexistentFilename('', 'foo', '.txt', 99)
+            })
+        })
+        it('returns a filename that does not exist in the directory', async () => {
+            const dir = tempFolder
+            await writeFile(path.join(dir, 'foo.txt'), '', 'utf8')
+            await writeFile(path.join(dir, 'foo-0.txt'), '', 'utf8')
+            await writeFile(path.join(dir, 'foo-1.txt'), '', 'utf8')
+            await writeFile(path.join(dir, 'foo-2.txt'), '', 'utf8')
+            assert.strictEqual(getNonexistentFilename(dir, 'foo', '.txt', 99), 'foo-3.txt')
+            assert.strictEqual(getNonexistentFilename(dir, 'foo', '', 99), 'foo')
+        })
+        it('returns "foo-RANDOM.txt" if max is reached', async () => {
+            const dir = tempFolder
+            await writeFile(path.join(dir, 'foo.txt'), '', 'utf8')
+            await writeFile(path.join(dir, 'foo-1.txt'), '', 'utf8')
+            // Looks like "foo-75446d5d.txt".
+            assert.ok(/foo-[a-fA-F0-9]{8}/.test(getNonexistentFilename(dir, 'foo', '.txt', 1)))
+        })
+    })
 
-    describe('makeTemporaryToolkitFolder', () => {
+    describe('makeTemporaryToolkitFolder()', () => {
```

Comment:
Why add the parenthesis to the test name?

Reply:
parens are a useful visual aid that explains that this is a function

Reply:
I'll allow it if we want to make this the custom going forward. It's just out-of-place in its current form.

Reply:
A small reason in favor of the "()" is that it signals to the next person that this is a symbol, not a description.

In this case, [this line](https://github.com/aws/aws-toolkit-vscode/blob/bcfbacbeb78b7069cd515187367d062330e3776d/src/test/shared/filesystemUtilities.test.ts#L91) already has the "()" so let's leave this PR as-is for now.